src/Windows.Core/PinnedStruct{T}.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Copyright (c) to owners found in https://github.com/AArnott/pinvoke/blob/master/COPYRIGHT.md. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+
+namespace PInvoke
+{
+    using System;
+    using System.Runtime.InteropServices;
+
+    /// <summary>
+    /// Allocate a structure on the heap and pin it in place, allowing a controlled release (Via Dispose).
+    /// </summary>
+    /// <typeparam name="T">Type of the structure that will be pinned.</typeparam>
+    public class PinnedStruct<T> : IDisposable
+        where T : struct
+    {
+        private readonly IntPtr intPtr;
+
+        private bool disposed;
+        private GCHandle? gcHandle;
+
+        public PinnedStruct(T? value)
+        {
+            if (value.HasValue)
+            {
+                object box = value.Value;
+                this.gcHandle = GCHandle.Alloc(box, GCHandleType.Pinned);
+                this.intPtr = this.gcHandle.Value.AddrOfPinnedObject();
+            }
+            else
```

Comment:
I would lose the else block here. In .NET, a class's fields are initialized to their default values automatically.